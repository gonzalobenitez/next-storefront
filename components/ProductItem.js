import { FormattedNumber } from 'react-intl'
import Link from 'next/link'

export default ({ product, add }) => {
  return (
  	<div className="product-item" >
	    <div className="overlay-container">
	    	<img src={ `/static/${ product.image }` } />
	    	<div className="overlay">
			    <button onClick={ add }>Add to Cart</button>
			    <Link href={{ pathname: '/product', query: { id: product.id } }}><button>View Details</button></Link>
	    	</div>
	    </div>
	    <span>{ product.brand }</span>
	    <h4>{ product.title }</h4>
	    <span className="price"><FormattedNumber value={ product.price } style="currency" currency={ 'USD' } /></span>

		<style jsx>{`
			.product-item {
				display: inline-block;
				margin: 0.75em;
				width: 100%;
				max-width: 18em;
				position: relative;
			}

			.product-item .overlay-container {
				position: relative;
				margin-bottom: 0.5em;
			}

			.product-item .overlay-container .overlay {
				display: none;
			}

			.product-item h4 {
				font-size: 0.8em;
				text-transform: uppercase;
				padding: 10px 0;
			}

			.product-item .price {
				display: block;
				padding-bottom: 10px;
			}

			.product-item .overlay-container img {
				width: 18em;
				height: 18em;
				position: relative;
				overflow: hidden;
				cursor: pointer;
				transition: box-shadow 0.25s;
			}

			@media(max-width: 600px) {
				.product-item .overlay-container img {
					width: 100vw;
					height: 100vw;
				}
			}

			.product-item:hover .overlay-container .overlay {
				display: block;
				position: absolute;
				background-color:rgba(0, 0, 0, 0.5);
				top: 0;
				left: 0;
				width: 100%;
				height: 100%;
				font-size: 5em;
				color: white;
				text-shadow: 0 0 5px rgba(0,0,0,0.5), 0 0 10px rgba(0,0,0,0.5); 
				transition: all 0.25s;
			}

			.product-item:hover {
				box-shadow: 0 12px 15px 0 rgba(0,0,0,0.24),0 17px 50px 0 rgba(0,0,0,0.19);
			}
		`}</style>
    </div>
  )
}
