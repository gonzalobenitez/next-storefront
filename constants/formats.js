export default {
  number: {
    USD: {
      style: 'currency',
      currency: 'USD'
    }
  }
}
