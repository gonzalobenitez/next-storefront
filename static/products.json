[
  {
    "title": "Blue Stripe Stoneware Plate",
    "brand": "Kiriko",
    "price": 40,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "blue-stripe-stoneware-plate.jpg",
    "id": 12
  },
  {
    "title": "Hand Painted Blue Flat Dish",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    "image": "hand-painted-blue-flat-dish.jpg",
    "id": 22
  },
  {
    "title": "Heme",
    "brand": "Dust & Form",
    "price": 52,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    "image": "heme.jpg",
    "id": 32
  },
  {
    "title": "Mashiko-Yaki Green Small Plate",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-green-small-plate.jpg",
    "id": 42
  },
  {
    "title": "Mashiko-Yaki Indigo Small Plate",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-indigo-small-plate.jpg",
    "id": 52
  },
  {
    "title": "Mashiko-Yaki Saucer",
    "brand": "Kiriko",
    "price": 18,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-saucer.jpg",
    "id": 62
  }
]
