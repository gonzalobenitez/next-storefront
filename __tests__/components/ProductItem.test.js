import { shallow } from 'enzyme'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import ProductItem from '../../components/ProductItem.js'
import products from '../../data/products.json'
import formats from '../../constants/formats'
import Store from '../../stores/store'

describe('ProductItem Test', () => {
  let store = null
  beforeEach(() => {
    store = new Store(products)
  })

  it('shows item', () => {
    const product = store.products[0]
    const shallowComponent = shallow(
      <ProductItem 
        product={ product }
        key={ product.id }
        add={() => store.addToCart(product.id, 1) }
      />
    )
    expect(shallowComponent.find('button').text()).toEqual('Add to Cart')
  })

  it('shows item in Snapshot Testing', () => {
    const product = store.products[0]
    const shallowComponent = renderer.create(
      <IntlProvider locale="en" formats={ formats } defaultFormats={ formats } >
        <ProductItem 
          product={ product }
          key={ product.id }
          add={() => store.addToCart(product.id, 1) }
        />
      </IntlProvider>
    )
    const tree = shallowComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
