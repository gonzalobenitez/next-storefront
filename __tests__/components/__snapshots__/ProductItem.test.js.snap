// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem Test shows item in Snapshot Testing 1`] = `
<div
  className="jsx-3577587527 product-item"
>
  <div
    className="jsx-3577587527 overlay-container"
  >
    <div
      className="jsx-3577587527 product-img"
    />
    <div
      className="jsx-3577587527 overlay"
    >
      <a
        className="jsx-3577587527 button"
        href="/product?id=12"
        onClick={[Function]}
      >
        View Details
      </a>
      <button
        className="jsx-3577587527"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </div>
  </div>
  <span
    className="jsx-3577587527"
  >
    Kiriko
  </span>
  <h4
    className="jsx-3577587527"
  >
    Blue Stripe Stoneware Plate
  </h4>
  <span
    className="jsx-3577587527 price"
  >
    <span>
      $40.00
    </span>
  </span>
</div>
`;
